cmds/boot/boot.go

```diff
@@ -0,0 +1,451 @@
+	if ( err != nil ) {	
+		return dest, err
+	}
+
+	destFile, err := os.Create(dest) // creates if file doesn't exist
+	if ( err != nil ) {
+		return dest, err
+        }
+
+	_, err = io.Copy(destFile, srcFile) // check first var for number of bytes copied
+	if ( err != nil ) {
+		return dest, err
+        }
+
+	err = destFile.Sync()
+	if ( err != nil ) {
+		return dest, err
+        }
+	err = destFile.Close()
+	if ( err != nil ) {
+		return dest, err
+        }
+	err = srcFile.Close()
+	if ( err != nil ) {
+		return dest, err
+        }
+	return dest, nil
+}
+
+// kexecEntry is booting new kernel based on the content of grub.cfg
+func kexecEntry(grubConfPath string, mountPoint string) error {
+	var fileMenuContent []string
+	var entry int
+	var localKernelPath string
+	var localInitrdPath string
+	if verbose {
+		log.Printf(grubConfPath)
+	}
+	fileMenuContent, entry, err := getFileMenuContent(grubConfPath)
+	if ( err != nil ) {
+		return err
+	}
+	var kernel string
+	var kernelParameter string
+	var initrd string
+	var kernelInfos []string
+	kernelInfos = strings.Fields(fileMenuContent[3*entry+1])
+	kernel = kernelInfos[1]
+	var count int
+	count = 0
+	for _, field := range kernelInfos {
+		if count > 1 {
+			kernelParameter = kernelParameter + " " + field
+		}
+		count = count + 1
+	}
+	fmt.Sscanf(fileMenuContent[3*entry+2], "initrd %s", &initrd)
+	if verbose {
+		log.Printf("************** boot parameters  ********************")
+		log.Printf(kernel)
+		log.Printf(kernelParameter)
+		log.Printf(initrd)
+		log.Printf("****************************************************")
+	}
+	localKernelPath,err = copyLocal(mountPoint + kernel)
+	if ( err != nil ) {
+		return err
+	}
+	localInitrdPath,err = copyLocal(mountPoint + initrd)
+	if ( err != nil ) {
+		return err
+	}
+	if verbose {
+		log.Printf(localKernelPath)
+	}
+	umountEntry(mountPoint)
+	// We can kexec the kernel with localKernelPath as kernel entry, kernelParameter as parameter and initrd as initrd !
+	log.Printf("Loading %s for kernel\n", localKernelPath)
+
+	kernelDesc, err := os.OpenFile(localKernelPath, os.O_RDONLY, 0)
+	if err != nil {
+		return err
+	}
+	// defer kernelDesc.Close()
+
+	var ramfs *os.File
+	ramfs, err = os.OpenFile(localInitrdPath, os.O_RDONLY, 0)
+	if err != nil {
+		return err
+	}
+	// defer ramfs.Close()
+
+	if err := kexec.FileLoad(kernelDesc, ramfs, kernelParameter); err != nil {
+		return err
+	}
+ 	err = ramfs.Close()
+	if ( err != nil ) {
+		return err
+	}
+	err = kernelDesc.Close()
+	if ( err != nil ) {
+		return err
+	}
+	if err := kexec.Reboot(); err != nil {
+		return err
+	}
+	return err
+
+}
+
+func registerFlags(f *flag.FlagSet) *options {
+        o := &options{}
+        f.BoolVar(&o.verbose, "v", false, "Set verbose output")
+        return o
+}
+
+
+func main() {
+	var blkList []string
+	var supportedFilesystem []string
+	verbose = false
+        opts := registerFlags(flag.CommandLine)
+	flag.Parse()
+
+        if opts.verbose != false {
```

Comment:
If you do what I'm suggesting you can remove these three lines

Reply:
Done